packages/music/src/classes/Queue.ts

```diff
@@ -632,4 +632,50 @@ export abstract class Queue<T extends Player = Player> {
     this.processQueue();
     return tracks;
   }
+  /**
+   * Play apple
+   * @param search
+   * @param options
+   * @param enqueueTop
+   * @returns
+   */
+   public async apple(
+    search: string,
+    options?: ITrackOptions,
+    enqueueTop?: boolean
+  ): Promise<YoutubeTrack[] | undefined> {
+    const appleTracks =
+      typeof search === "string" ? await Util.getAppleTracks(search) : search;
+    if (!appleTracks) {
+      return;
+    }
+    let allVideos: (Video | undefined)[] = [];
+    if (appleTracks.type === "song") {
+        const song = await Util.getSong(
+            appleTracks.title +
+            " - " +
+            appleTracks.artist
+        );
+        allVideos = [song];
```

Comment:
you are not collecting all videos, and rewriting it.

Reply:
wdym sorry